src/Nethermind/Nethermind.Trie/TrieNode.cs

```diff
@@ -507,7 +509,10 @@ public void ResolveKey(ITrieNodeResolver tree, ref TreePath path, bool isRoot, I
             if (rlp is null || IsDirty)
             {
                 ref readonly CappedArray<byte> oldRlp = ref rlp is not null ? ref rlp.Data : ref CappedArray<byte>.Empty;
-                CappedArray<byte> fullRlp = RlpEncode(tree, ref path, bufferPool);
+                CappedArray<byte> fullRlp = NodeType == NodeType.Branch ?
```

Comment:
RlpEncode contains this check already, does it optimize something?

Reply:
Removes about 11 stack frames since Branch is the most common and RlpEncode doesn't inline or tailcall because the return is larger than a pointer size

Reply:
Will try to get a trace that shows what it looks like now

Reply:
After PR

<img width="509" alt="image" src="https://github.com/NethermindEth/nethermind/assets/1142958/8995a267-0908-4de5-968f-040697a86d39">

Before it has 3 more methods on each recursion